src/python/CMakeLists.txt

```diff
@@ -78,6 +85,67 @@ install(FILES "${OpenVINOGenAI_SOURCE_DIR}/LICENSE"
 
 # wheel_genai component is used for wheel generation in pyproject.toml.
 # Exclude wheel_genai from normal packaging because there's pygenai_X_Y component for that.
-install(TARGETS openvino_genai py_openvino_genai
+install(TARGETS openvino_genai ${TARGET_NAME}
         LIBRARY DESTINATION openvino_genai COMPONENT wheel_genai EXCLUDE_FROM_ALL
         RUNTIME DESTINATION openvino_genai COMPONENT wheel_genai EXCLUDE_FROM_ALL)
+
+# Generate or check .pyi stub files generated by pybind11-stub
+
+set(pyproject_toml "${OpenVINOGenAI_SOURCE_DIR}/pyproject.toml")
+file(STRINGS ${pyproject_toml} pybind11_stubgen_dep REGEX "pybind11-stubgen")
+
+if(pybind11_stubgen_dep MATCHES "pybind11-stubgen==[0-9\.]+")
+    set(pybind11_stubgen_dep "${CMAKE_MATCH_0}")
+else()
+    message(FATAL_ERROR "Internal error: failed to parse pybind11-stubgen version from from '${pyproject_toml}'")
+endif()
+
+if(OpenVINODeveloperPackage_FOUND)
+    set(requirements_file "${OpenVINOGenAI_SOURCE_DIR}/requirements-build.txt")
+
+    ov_check_pip_package(REQUIREMENT ${pybind11_stubgen_dep}
+                         RESULT_VAR pybind11_stubgen_AVAILABLE
+                         WARNING_MESSAGE "Please, install ${pybind11_stubgen_dep} if you plan to develop Python OpenVINO GenAI API"
+                         MESSAGE_MODE WARNING)
+elseif(DEFINED PY_BUILD_CMAKE_PACKAGE_NAME)
+    # in case of wheel build, pybind11-stubgen is always available via pyproject.toml's build-system
+    set(pybind11_stubgen_AVAILABLE ON)
+endif()
+
+if(pybind11_stubgen_AVAILABLE)
+    if(DEFINED ENV{CI} OR DEFINED ENV{TF_BUILD} OR DEFINED ENV{JENKINS_URL})
+        set(ci_run ON)
+    endif()
+
+    if(ci_run)
+        set(stub_files_location "${OpenVINOGenAI_BINARY_DIR}/src/python")
+    else()
+        set(stub_files_location "${OpenVINOGenAI_SOURCE_DIR}/src/python")
+    endif()
+    set(output_files "${stub_files_location}/openvino_genai/py_openvino_genai.pyi"
+                     "${stub_files_location}/openvino_genai/__init__.pyi")
+
+    add_custom_command(OUTPUT ${output_files}
+        COMMAND "${CMAKE_COMMAND}" -E env PYTHONPATH=${CMAKE_BINARY_DIR}
+                "${Python3_EXECUTABLE}" -m pybind11_stubgen --output-dir "${stub_files_location}" openvino_genai
+        COMMAND
+                "${CMAKE_COMMAND}"
+                    -D generated_pyi_files_location=${stub_files_location}
+                    -D source_pyi_files_location=${OpenVINOGenAI_SOURCE_DIR}/src/python
+                    -P "${CMAKE_CURRENT_SOURCE_DIR}/compare_pyi.cmake"
+        DEPENDS
+            ${python_sources}
+            "${CMAKE_CURRENT_SOURCE_DIR}/openvino_genai/__init__.py"
+            "${CMAKE_CURRENT_SOURCE_DIR}/compare_pyi.cmake"
+        COMMENT "[${pybind11_stubgen_dep}] Generate .pyi files"
+        VERBATIM)
+
+    add_custom_target(${TARGET_NAME}_stub ALL DEPENDS ${output_files})
```

Comment:
py_openvino_genai_stub must always depend on src .pyi.

1. I run `export CI=1`
2. build passed
3. corrupted src `py_openvino_genai.pyi`
4. build failed
5. restored src `py_openvino_genai.pyi`
6. build failed with the same message

Reply:
please, check once again